api/openai/request.go

```diff
@@ -15,6 +16,9 @@ import (
 
 func readInput(c *fiber.Ctx, loader *model.ModelLoader, randomModel bool) (string, *OpenAIRequest, error) {
 	input := new(OpenAIRequest)
+	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
use the context that comes from the app here instead of `context.Background`, so there is still control of it from the top-level AppOptions

Reply:
good call, done!